packages/core/src/lib/TwaManifest.ts

```diff
@@ -315,6 +308,102 @@ export class TwaManifest {
     const json = JSON.parse((await fs.promises.readFile(fileName)).toString());
     return new TwaManifest(json);
   }
+
+  /**
+   * Given a field name, returns the new value of the field.
+   *
+   * @param {string} fieldName the name of the given field.
+   * @param {string[]} fieldsToIgnore the fields which needs to be ignored.
+   * @param {T} oldValue the old value of the field.
+   * @param {T} newValue the new value of the field.
+   * @returns {T}
+   */
+  static getNewFieldValue<T>(fieldName: string, fieldsToIgnore: string[],
+      oldValue: T, newValue: T): T {
+    if (fieldsToIgnore.includes(fieldName)) {
+      return oldValue;
+    }
+    return newValue || oldValue;
+  }
+
+  /**
+   * Gets the shortcuts from the web manifest.
+   *
+   * @param {URL} webManifestUrl the URL where the webManifest is available.
+   * @param {WebManifest} webManifest the Web Manifest.
+   * @returns {ShortcutInfo[]}
+   */
+  static getShortcuts(webManifestUrl: URL, webManifest: WebManifestJson): ShortcutInfo[] {
+    const shortcuts: ShortcutInfo[] = [];
+    for (let i = 0; i < (webManifest.shortcuts || []).length; i++) {
+      const s = webManifest.shortcuts![i];
+      try {
+        const shortcutInfo = ShortcutInfo.fromShortcutJson(webManifestUrl, s);
+        if (shortcutInfo != null) {
+          shortcuts.push(shortcutInfo);
+        }
+      } catch (err) {
+        TwaManifest.log.warn(`Skipping shortcut[${i}] for ${err.message}.`);
+      }
+      if (shortcuts.length === 4) {
+        break;
+      }
+    }
+    return shortcuts;
+  }
+
+  /**
+   * Merges the Twa Manifest with the web manifest. Ignores the specified fields.
+   *
+   * @param {string[]} fieldsToIgnore the fields which needs to be ignored.
+   * @param {URL} webManifestUrl the URL where the webManifest is available.
+   * @param {WebManifest} webManifest the Web Manifest, used as a base for the update of
+   *    the TWA Manifest.
+   * @param {TwaManifest} oldTwaManifest current Twa Manifest.
+   * @returns {Promise<TwaManifest>} the new and merged Twa manifest.
+   */
+  static async merge(fieldsToIgnore: string[], webManifestUrl: URL,
+      webManifest: WebManifestJson, oldTwaManifest: TwaManifest): Promise<TwaManifest> {
+    function resolveIconUrl(icon: WebManifestIcon | null): string | undefined {
```

Comment:
You could make this method `updateIconUrl` and do:

```
function updateIconUrl(icon: WebManifestIcon | null, oldUrl: string) : string {
    return icon ? new URL(icon.src, webManifestUrl).toString() : oldUrl;
}
```
And then it would simplify lines 400-402.

Reply:
Actually, you could also have it include the call to `findSuitableIcon`.

Reply:
I think that calling `findSuitableIcon` will not do much, I still need to pass all of the arguments so the code will be the same

Reply:
Would there be less code duplication if we called `findSuitableIcon`?

Reply:
I think that the latest change solved the problem, let me know what do you think